### PR TITLE
NO-ISSUE: imagebasedinstaller:  support json files in extramanifests directory

### LINKED ISSUE #9134: IBI Config image does not include .json files from the extra-manifests directory
# Version

```console
$ openshift-install version
openshift-install417 4.17.0-rc.5
built from commit dfd4c085a7210e49111fa8d6747016d78f98ecf2
release image quay.io/openshift-release-dev/ocp-release@sha256:5b1399dc8175d7d26611c2541962d101948cba6a951ed3dd51cddf75afbab303
```

# Platform:
baremetal
* IBI (Image Based Installer)

# What happened?

When creating an IBI Config image, that contains an `extra-manifests` directory, `.json` files are ignored and not added to the config ISO.

# What you expected to happen?

All appropriate configuration files including YAML, YML and JSON should be included and applied at configuration time.

# How to reproduce it (as minimally and precisely as possible)?

1. create `install` directory with a subdirectory of `extra-manifests`
2. copy in a install-config.yaml and image-based-config.yaml file to the install directory
3. create a test.yaml and test.json and place in `install\extra-manifests`
4. run `openshift-install417 image-based create config-image --dir install/`
5. review the contents of the install/imagebasedconfig.iso and find that `test.json` is NOT included in `extra-manifests` on the ISO image

# Anything else we need to know?

The problem exists here: https://github.com/openshift/installer/blob/master/pkg/asset/imagebased/configimage/extramanifests.go#L42

I am working on a PR now that addresses this issue, and will submit it later today. I will be sure to reference this issue in the PR request.

# References

### PR DESCRIPTION
**Pull request by xphyr**
Added support for `.json` files in the extra-manifests directory when creating a config-image using the "image-based" installer. Updated test cases as well to test added functionality.

Fixes #9134

### CODE REVIEW

**Review by mresvanis — COMMENTED** [review]
Another optional improvement would be to add an image-based integration [test](https://github.com/openshift/installer/tree/master/cmd/openshift-install/testdata/imagebased) verifying that all the provided extra-manifests are included properly in the configuration ISO.

Integration tests are based on [testscript](https://github.com/rogpeppe/go-internal/tree/master?tab=readme-ov-file#testscript). Specifically, I would add a `cmd/openshift-install/testdata/imagebased/configimage/manifests/extra_manifests.txt`, which would look like the [configimage/assets/default_assets.txt](https://github.com/openshift/installer/blob/master/cmd/openshift-install/testdata/imagebased/configimage/assets/default_assets.txt). But it would include, in addition to the config files, the `extra-manifests/{test.yaml,test.yml,test.json}` in the `WORKDIR` and verify that those are included in the resulting ISO.

As I mentioned that's absolutely optional, I thought this would be a useful addition to the integration tests, as we have so far implemented very basic integration test cases. WDYT?

File: cmd/openshift-install/testdata/imagebased/configimage/manifests/extra_manifests.txt [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Verify that the extra-manifests directory populates with the appropriate files
+
+exec openshift-install image-based create config-image --dir $WORK
+
+exists $WORK/imagebasedconfig.iso
+exists $WORK/auth/kubeconfig
```
Comment (mresvanis) [inline cmd/openshift-install/testdata/imagebased/configimage/manifests/extra_manifests.txt:6]: We can get rid of the checks for the `auth/` and the `/cluster-configuration/manifest.json`, as those are tested in the `assets/default_assets.txt` integration test.

File: cmd/openshift-install/testdata/imagebased/configimage/manifests/extra_manifests.txt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+apiVersion: v1beta1
+kind: DummyData
+metadata:
+  name: test.yaml
+
+-- extra-manifests/test.yml --
+apiVersion: v1beta1
+kind: DummyData
+metadata:
+  name: test.yml
```
Comment (eranco74) [inline cmd/openshift-install/testdata/imagebased/configimage/manifests/extra_manifests.txt:54]: one of these should be `another-test.yml`, no?

**Review by eranco74 — COMMENTED** [review]
Seems that it should work on the LCA side
https://github.com/openshift-kni/lifecycle-agent/blob/18bcf604ca831c2676c33ed02e7ad21ae1c7ba8c/lca-cli/postpivot/postpivot.go#L397

File: cmd/openshift-install/testdata/imagebased/configimage/manifests/extra_manifests.txt (hunk 2 above)
Comment (mresvanis) [reply cmd/openshift-install/testdata/imagebased/configimage/manifests/extra_manifests.txt:54]: I think `test.yml` is enough, as this here is the input file to the installer (and `metadata.name` contains the suffix). In the end, we have one manifests for each pattern, `.yaml`, `.yml`, `.json`, were you thinking about an additional test case (that I'm missing)?

File: pkg/asset/imagebased/configimage/extramanifests_test.go [hunk 3]
```diff
@@ -112,6 +131,18 @@ func TestExtraManifests_Load(t *testing.T) {
 				} else {
 					fileFetcher.EXPECT().FetchByPattern("extra-manifests/*.yml").Return(ymlFiles, nil)
 				}
+			} else {
```
[the remark this answers is not in the document]
Comment (mresvanis) [reply pkg/asset/imagebased/configimage/extramanifests_test.go]: But then how dow we setup the fileFetcher expected call for the `*.yml` files , i.e. it's going to be called and we need to see if it needs to throw an error or nil?

File: cmd/openshift-install/testdata/imagebased/configimage/manifests/extra_manifests.txt (hunk 2 above)
Comment (mresvanis) [reply cmd/openshift-install/testdata/imagebased/configimage/manifests/extra_manifests.txt:54]: Oh but this here is for the intergation tests, there is no connection to the unit tests. There is no technical need for those files to have the same names.

File: pkg/asset/imagebased/configimage/extramanifests_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (eranco74) [reply pkg/asset/imagebased/configimage/extramanifests_test.go]: :man_facepalming: 
I missed that it's handling a different suffix

**Pull request merged**